core/src/main/java/jenkins/I18n.java

```diff
@@ -93,7 +93,16 @@ public HttpResponse doResourceBundle(StaplerRequest request) {
         String language = request.getParameter("language");
         String country = request.getParameter("country");
         String variant = request.getParameter("variant");
-
+        // https://www.w3.org/International/questions/qa-lang-priorities
+        // in case we have regions/countries in the language query parameter
+        if (country == null && language.length()>2) {
+            country = language.substring(3,5);
+            // not mentioned in above url but we assume variants as en-US-Cloudbees
+            if (variant == null && language.length()>5) {
+               variant = language.substring(6);
```

Comment:
https://docs.oracle.com/javase/7/docs/api/java/util/Locale.html 

```
variant
Any arbitrary value used to indicate a variation of a Locale. Where there are two or more variant values each indicating its own semantics, these values should be ordered by importance, with most important first, separated by underscore('_'). The variant field is case sensitive.
```

In the test case I am requesting ```language=de-DE-cloudBees``` now if there would be a file ```hudson.logging.Messages_de_DE_cloudBees``` we would request that one.